### PR TITLE
tests: add e2e tests for plugin

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
test

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
NA

**Summary**
- Add e2e tests for plugin

**Does this PR introduce a breaking change?**
No

**Other information**
- Any other cases to be tested are appreciated.

Did some util changes in #1476 , this should go after that.

### CODE REVIEW

**Comment by anshumanv** [comment]
Ahh, looking at tests

File: packages/webpack-cli/__tests__/init/auto/init-auto.test.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe('init auto flag', () => {
         expect(stdout).not.toContain(firstPrompt);
 
         // Test regressively files are scaffolded
-        const files = ['./sw.js', './package.json', './yarn.lock', './src/index.js'];
```
Comment (anshumanv) [inline packages/webpack-cli/__tests__/init/auto/init-auto.test.js:25]: For some reason this was failing when we're checking for lock file,

File: test/loader/loader.test.js [hunk 2]
```diff
@@ -29,10 +29,10 @@ describe('loader command', () => {
         expect(existsSync(join(__dirname, loaderName))).toBeTruthy();
 
         // Test regressively files are scaffolded
-        const files = ['package.json', 'yarn.lock', 'examples', 'src', 'test', 'src/index.js', 'examples/simple/webpack.config.js'];
```
Comment (anshumanv) [inline test/loader/loader.test.js:32]: Same reason, checking for lock file is failing at random instances, I think before the file is actually created we're running a check for it.

**Comment by evilebottnawi** [comment]
@anshumanv strange, CI green

**Comment by anshumanv** [comment]
@evilebottnawi I removed checks for yarn.lock, fails otherwise 😭

**Comment by evilebottnawi** [comment]
@anshumanv problems only locally?

**Comment by anshumanv** [comment]
@evilebottnawi only in MacOS CI, passes locally on MacOS

**Comment by anshumanv** [comment]
> Instead of yarn.lock, you can check for existence of node_modules folder. yarn.lock is there for checking whether packages are installed or not by the generator in the end. I think it would help for now.

Let me see if that works. 👍

File: test/init/auto/init-auto.test.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ describe('init auto flag', () => {
         expect(stdout).not.toContain(firstPrompt);
 
         // Test regressively files are scaffolded
-        const files = ['./sw.js', './package.json', './yarn.lock', './src/index.js'];
+        const files = ['./sw.js', './package.json', './src/index.js'];
```
Comment (rishabh3112) [inline test/init/auto/init-auto.test.js:36]: This change is not needed now.

File: test/init/generator/init-inquirer.test.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ describe('init', () => {
         expect(stdout).toContain(firstPrompt);
 
         // Test regressively files are scaffolded
-        const files = ['./sw.js', './package.json', './yarn.lock', './src/index.js'];
+        const files = ['./sw.js', './package.json', './src/index.js'];
```
Comment (rishabh3112) [inline test/init/generator/init-inquirer.test.js:31]: Same here

File: test/init/auto/init-auto.test.js (hunk 3 above)
Comment (anshumanv) [reply test/init/auto/init-auto.test.js:36]: Needed, these are still randomly failing.

File: test/init/generator/init-inquirer.test.js (hunk 4 above)
Comment (anshumanv) [reply test/init/generator/init-inquirer.test.js:31]: ^ Same

File: test/init/auto/init-auto.test.js (hunk 3 above)
Comment (rishabh3112) [reply test/init/auto/init-auto.test.js:36]: Then add a check for `node_modules` folder, we need to check whether packages are installed or not.

Comment (anshumanv) [reply test/init/auto/init-auto.test.js]: Done, I hope it stays consistent now.

Comment (anshumanv) [reply test/init/auto/init-auto.test.js]: Tried adding node_modules, same behaviour, would need debugging later.

File: test/plugin/plugin.test.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+const rimraf = require('rimraf');
+const { run, runPromptWithAnswers } = require('../utils/test-utils');
+
+const ENTER = '\x0D';
+const firstPrompt = '? Plugin name';
+const pluginName = 'test-plugin';
+const pluginPath = join(__dirname, pluginName);
+
+// Since scaffolding is time consuming
+jest.setTimeout(200000);
```
Comment (evilebottnawi) [inline test/plugin/plugin.test.js:12]: Why number is so big?

Comment (anshumanv) [reply test/plugin/plugin.test.js:12]: I'll reduce this, maybe I was testing

Comment (anshumanv) [reply test/plugin/plugin.test.js]: Fixed @evilebottnawi

**Pull request merged**